scripts/core/superdesk-api.d.ts

```diff
@@ -105,6 +105,7 @@ declare module 'superdesk-api' {
             monitoring?: {
                 getFilteringButtons?(deskId: string): Promise<Array<IMonitoringFilter>>;
             };
+            configurableUiComponents?: IConfigurableUiComponents;
```

Comment:
The idea behind configurable UI components is that only one of the same type should be allowed per instance. That's why I'm registering them in [startApp](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/index.ts#L112) and not allowing extensions to provide them.

I've already implemented an API for side widgets and I suggest you use that(and disable the original one if the one from extensions is present). It's not merged to develop yet, but you can merge it to `sams` branch since it won't change much, except for styles.

Reply:
Here's the PR https://github.com/superdesk/superdesk-client-core/pull/3575/files#diff-77f7ccd12c27cd27aa6bdedde8c6be34R80

Reply:
I have reverted to use the intended method of ConfigurableUiComponents and removed the ability for extensions to provide these.
Attachments can be used in two places, the sidebar widget and as a field in the editor. Because of this, I will continue to use the ConfigurableUiComponents functionality and let the instance provide he config for the intended component to use.

I have also removed the usage of `redux` for attachments, as it was only really used to store the list of files. The new HOC now provides the API via props

Reply:
> Attachments can be used in two places, the sidebar widget and as a field in the editor.

You could also expose the same component to multiple extension points. I've done that with "mark for user" widget for example. There's also an [extension point for article fields](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/superdesk-api.d.ts#L72) in the API.

Reply:
This field/widget replaces an existing one though, not adding a new one. Displaying and providing very similar experience to the one that's being replaced. ConfigurableUiComponents fits that description quite well.